pkg/pgclient/metrics.go

```diff
@@ -5,73 +5,65 @@
 package pgclient
 
 import (
-	"math"
-
+	"github.com/jackc/pgx/v4/pgxpool"
 	"github.com/prometheus/client_golang/prometheus"
+
 	"github.com/timescale/promscale/pkg/util"
 )
 
-var (
-	statementCacheLen prometheus.Histogram
-	statementCacheCap = prometheus.NewGauge(
+func initMetrics(r prometheus.Registerer, writerPool, readerPool *pgxpool.Pool) {
+	writerAcquired := prometheus.NewGaugeFunc(
 		prometheus.GaugeOpts{
-			Namespace: util.PromNamespace,
-			Subsystem: "cache",
-			Name:      "statement_per_connection_capacity",
-			Help:      "Maximum number of statements in connection pool's statement cache",
+			Namespace:   util.PromNamespace,
+			Subsystem:   "sql_database",
+			Name:        "acquired_connections",
+			Help:        "Number of connections currently acquired from the pool.",
+			ConstLabels: map[string]string{"pool": "writer"},
+		}, func() float64 {
+			if writerPool == nil {
+				// readonly mode.
+				return 0
+			}
+			return float64(writerPool.Stat().AcquiredConns())
 		},
 	)
-	statementCacheEnabled = prometheus.NewGauge(
+	writerActive := prometheus.NewGaugeFunc(
 		prometheus.GaugeOpts{
 			Namespace:   util.PromNamespace,
-			Subsystem:   "cache",
-			Name:        "enabled",
-			Help:        "Cache is enabled or not.",
-			ConstLabels: map[string]string{"type": "metric", "name": "statement_cache"},
+			Subsystem:   "sql_database",
+			Name:        "active_connections",
```

Comment:
This connections are all the connection in the pool. Can be active + idle. So maybe better naming `total_connections`?